### ISSUE TITLE
Function arguments based on `#[cfg(feature = "xyz")]` are handled incorrectly

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#6148]
## Environment

* **IntelliJ Rust plugin version:** 0.3.132.3360-202-nightly
* **Rust toolchain version:** 1.46.0 (04488afe3 2020-08-24) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2020.2.2 Ultimate Edition (IU-202.7319.50)
* **Operating system:** Linux 5.4.0-47-generic

## Problem description

Consider this example:
```rust
fn main() {
    foo("123");
}

fn foo(#[cfg(windows)] a: i32, #[cfg(unix)] b: &str) {
    
}
```

It gives 2 false-positives:
1. mismatched types [E0308] expected `i32`, found `&str` 
2. This function takes 2 parameters but 1 parameter was supplied [E0061]

![image](https://user-images.githubusercontent.com/3221931/93808187-52832400-fc54-11ea-8c45-dcef0ef0d3e6.png)

Also, the function is weirdly highlighted:

![image](https://user-images.githubusercontent.com/3221931/93807967-0e901f00-fc54-11ea-9327-5f72780c91cb.png)
 
Similar to #6147
